winforms/src/toga_winforms/widgets/table.py

```diff
@@ -108,6 +111,48 @@ def winforms_cache_virtual_items(self, sender, e):
         for i in range(new_length):
             self._cache.append(self._new_item(i + self._first_item))
 
+    def winforms_search_for_virtual_item(self, sender, e):
+        if (
+            not e.IsTextSearch or not self._accessors or not self._data
+        ):  # pragma: no cover
+            # If winforms fires this event as a location based search,
+            # or with any invalid parameters, return gracefully.
+            return
+        find_previous = e.Direction in [
+            WinForms.SearchDirectionHint.Up,
+            WinForms.SearchDirectionHint.Left,
+        ]
+        i = e.StartIndex
+        found_item = False
+        while True:
+            # Either winforms might provide a starting index out of bounds if searching at list borders,
+            # or this loop may travel out of bounds itself while searching. In either case, wrap around.
+            if i < 0:  # pragma: no cover
+                # This could theoretically happen if this event is fired with a backwards search direction,
+                # however this edgecase should not take place within Toga's intended use of this event.
```

Comment:
Minor style tweak - we try to keep comments to ~80 chars, allowing up to 88 at a stretch if it helps with flow.

Reply:
Sorry for the delay, this one is stumping me because I very rarely contribute to projects coded by sighted people and thus I haven't run into this restriction before. I'll work out a way to stylize comments such as by writing a program that reads clipboard input, trims all whitespace then reports on the line length so I know whether the comment needs further shortening without spamming the right arrow key 80 times and slowly counting, removing one word then doing it again etc. I won't be able to write comments for Toga's codebase until I've come up with an accessible solution or another blind person tells me one, and none are around at the second.

Reply:
Sorry for double comment, just remembered to ask. Is this 80 chars per comment or 80 chars per line, I'm guessing per-comment?

Reply:
The limit is 80 characters per line of code, including any leading space. It's essentially a holdover from old teletype programming - you should be able to see the entire line of code on an 80x25 text mode terminal screen, with no line wrapping. We've now got much larger screens that aren't constrained by 80 columns, but visual readability decreases with line length, so it's now a "soft 80" limit.

The fact that this isn't an consideration for non-sighted programmers is a fascinating insight - thanks for letting me know about that.

As far as automated tooling goes - most sighted IDEs (e.g., Visual Studio) can do an "auto word-wrap" on comments; I'm not sure what your tools will allow. At the very least, the flake8 configuration *should* be catching this, but I've just noticed that our flake8 configuration sets a line length of 119 (the config is in tox.ini for... reasons), so we're not flagging > 88 character lines. That's something we should probably fix - I've opened #2975 to track this improvement.

In the meantime, I definitely don't want this to be an impediment to your ability to contribute to Toga. If you can't find an automated solution, I'm more than happy to do the last bit of leg work and re-flow long comments. I often end up tweaking comments in PRs anyway (purely for language style and content reasons), so it's not a major imposition.

Reply:
I thought we standardized on 88 rather than 80, for the reasons explained [here](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length)? It's better to be consistent, so we don't end up in the situation we had before when code was being reformatted every time it moved from one developer to another.

In VS Code I use [this extension](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap).